src/lambda/wizards/samInitWizard.ts

```diff
@@ -80,6 +100,171 @@ export class DefaultCreateNewSamAppWizardContext extends WizardContext implement
         return val ? (val.label as Runtime) : undefined
     }
 
+    public async promptUserForTemplate(
+        currRuntime: Runtime,
+        currTemplate?: SamTemplate
+    ): Promise<SamTemplate | undefined> {
+        const templates = getSamTemplateWizardOption(currRuntime)
+        const quickPick = picker.createQuickPick<vscode.QuickPickItem>({
+            options: {
+                ignoreFocusOut: true,
+                title: localize('AWS.samcli.initWizard.template.prompt', 'Select a SAM Application Template'),
+                value: currTemplate ? currTemplate : ''
+            },
+            buttons: [this.helpButton, vscode.QuickInputButtons.Back],
+            items: templates.toArray().map(template => ({
+                label: template,
+                alwaysShow: template === currTemplate,
+                description:
+                    template === currTemplate
+                        ? localize('AWS.wizard.selectedPreviously', 'Selected Previously')
+                        : getTemplateDescription(template)
+            }))
+        })
+
+        const choices = await picker.promptUser({
+            picker: quickPick,
+            onDidTriggerButton: (button, resolve, reject) => {
+                if (button === vscode.QuickInputButtons.Back) {
+                    resolve(undefined)
+                } else if (button === this.helpButton) {
+                    vscode.env.openExternal(vscode.Uri.parse(samInitDocUrl))
+                }
+            }
+        })
+        const val = picker.verifySinglePickerOutput(choices)
+
+        //eventBridgeStarterAppTemplate requires aws credentials
+        if (val && val.label === eventBridgeStarterAppTemplate) {
+            if (this.credentialsNotFound) {
+                await ext.awsContextCommands.onCommandLogin()
+
+                return exitWizard
+            }
+
+            //check if detected samCliVersion supports eventBridgeStarterAppTemplate
+            if (semver.lt(this.samCliVersion, MINIMUM_SAM_CLI_VERSION_SCHEMAS_SUPPORT_INCLUSIVE)) {
+                vscode.window.showErrorMessage(
+                    localize(
+                        'AWS.samcli.error.invalid_schema_support_version',
+                        'Installed SAM executable does not support templates that require Event Schema selection. Required minimum version {0}, but found {1}',
+                        MINIMUM_SAM_CLI_VERSION_SCHEMAS_SUPPORT_INCLUSIVE,
+                        this.samCliVersion
+                    )
+                )
+
+                return exitWizard
+            }
+        }
+
+        return val ? (val.label as SamTemplate) : undefined
+    }
+
+    public async promptUserForRegion(currRegion?: string): Promise<string | undefined> {
+        const schemasRegions = ['us-east-1', 'us-east-2', 'us-west-2', 'eu-west-1', 'ap-northeast-1']
```

Comment:
Please add a TODO comment here and reference #850 

I'm working on some changes that will allow us to get which regions a service is available in, and this is a perfect use case.

Reply:
That is awesome, I was planning on a separate pr after this pr to implement that feature, good to know it is already in the works.